### ISSUE TITLE
Index out of range exception in execute

### ISSUE THREAD

**Issue by misaqsaadat** [dotnet/machinelearning#5306]
### System information

windows 10 version 1803
NET Version (3.1.300): 
AutoML version : 0.17.1
(also tested on version 0.14.0)
### Issue

I am trying to use AutoML for the first time. but I get IndexOutOfRangeException on execute method call.

### Source code / logs
static void Main(string[] args)
		{
			MLContext mlContext = new MLContext();
			IDataView trainDataView = mlContext.Data.LoadFromTextFile<ModelInput>("input.csv", hasHeader: true);
			var experimentSettings = new RegressionExperimentSettings();
			experimentSettings.MaxExperimentTimeInSeconds = 100;
			RegressionExperiment experiment = mlContext.Auto().CreateRegressionExperiment(experimentSettings);
			ExperimentResult<RegressionMetrics> experimentResult = experiment.Execute(trainDataView, "Withdrawal");
			RegressionMetrics metrics = experimentResult.BestRun.ValidationMetrics;
			Console.WriteLine($"R-Squared: {metrics.RSquared:0.##}");
			Console.WriteLine($"Root Mean Squared Error: {metrics.RootMeanSquaredError:0.##}");
		}

Error:

Unhandled exception. System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
   at Microsoft.ML.AutoML.CrossValSummaryRunner`1.Run(SuggestedPipeline pipeline, DirectoryInfo modelDirectory, Int32 iterationNum)
   at Microsoft.ML.AutoML.Experiment`2.Execute()
   at Microsoft.ML.AutoML.ExperimentBase`2.Execute(ColumnInformation columnInfo, DatasetColumnInfo[] columns, IEstimator`1 preFeaturizer, IProgress`1 progressHandler, IRunner`1 runner)
   at Microsoft.ML.AutoML.ExperimentBase`2.ExecuteCrossValSummary(IDataView[] trainDatasets, ColumnInformation columnInfo, IDataView[] validationDatasets, IEstimator`1 preFeaturizer, IProgress`1 progressHandler)
   at Microsoft.ML.AutoML.ExperimentBase`2.Execute(IDataView trainData, ColumnInformation columnInformation, IEstimator`1 preFeaturizer, IProgress`1 progressHandler)
   at Microsoft.ML.AutoML.ExperimentBase`2.Execute(IDataView trainData, String labelColumnName, String samplingKeyColumn, IEstimator`1 preFeaturizer, IProgress`1 progressHandler)
   at FunWithMlnet.Program.Main(String[] args) in C:\Users\User\source\repos\FunWithWithdrawalsData\FunWithMlnet\Program.cs:line 18

**Comment by michaelgsharp** [comment]
This happens when you are trying to use a column that is not in your dataset. Could you share a sample of what your dataset looks like? Including the header?

**Comment by misaqsaadat** [comment]
[shorted.csv.txt](https://github.com/dotnet/machinelearning/files/4923337/shorted.csv.txt)

this is my input files. i remove some columns and some rows

**Comment by justinormont** [comment]
I would modify R2 calculation code to return `NaN` instead of `-Infinity` when there are <2 rows of data.

Our R2 calculation code:
https://github.com/dotnet/machinelearning/blob/beb67a128544ee98b8c4309859f7e5a9f679ac4e/src/Microsoft.ML.Data/Evaluators/RegressionEvaluator.cs#L101-L107

Returning `NaN` instead of `-Infinity` replicates the behavior of scikit:
https://github.com/scikit-learn/scikit-learn/blob/fd237278e895b42abe8d8d09105cbb82dc2cbba7/sklearn/metrics/_regression.py#L587-L590

**Comment by misaqsaadat** [comment]
sorry I don't get it. what's your mean less than two rows?
there are 20 rows is sample I uploaded and much more rows in my real case

**Comment by justinormont** [comment]
@misaqsaadat: My apologies, my recommendation was for how to fix the bug in ML․NET. Until it's fixed, you can mitigate the issue by using a larger dataset, or optimizing towards a different metric.
